contracts/src/Gateway.sol

```diff
@@ -141,6 +141,9 @@ contract Gateway is IGateway, IInitializable {
 
         // Dispatch message to a handler
         if (message.command == Command.AgentExecute) {
+            if (channel.mode == OperatingMode.RejectingInboundMessages) {
+                revert Disabled();
+            }
```

Comment:
Ron, we need to think about this a bit more. Since setting the mode of the outbound-queue pallet on BridgeHub to `RejectingOutboundMessages` can have the same effect.

So we need to analyse the difference in the approaches, and consider the following issues among others:
* Which one works better when used for upgrades.
* Which approach results in users losing messages/funds? If messages rejected at BridgeHub, or at the Gateway?
* Which one works better for emergency shutdowns?

I would suggest rather leaving this work for another PR until we have that analysis above.

Reply:
> Since setting the mode of the outbound-queue pallet on BridgeHub to RejectingOutboundMessages can have the same effect as setting the mode of InboundChannel to OperatingMode.RejectingInboundMessages

IMHO it's not an either-or situation. I prefer to close both entry points at the same time to achieve max security, concern based on some previous comment [here](https://github.com/Snowfork/snowbridge/pull/882#discussion_r1274840347).

And I'm fine to move it out to make this PR more focused.

Reply:
Yes, lets please move it out.

I added a comment on https://github.com/Snowfork/snowbridge/pull/882#discussion_r1274840347

> Assuming there a bug in code, there is no way to guard against this kind of attack, since if a malicious attacker can manually construct a transaction and have it pass verification by the Gateway, then they can take over entire control of the Gateway using the upgrade handler.